src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -2020,11 +2020,28 @@ internal Task<Compilation> GetCompilationAsync(Project project, CancellationToke
                 : SpecializedTasks.Default<Compilation>();
         }
 
-        internal Task<bool> HasCompleteReferencesAsync(Project project, CancellationToken cancellationToken)
+        /// <summary>
+        /// Return reference completeness for the given project.
+        /// </summary>
+        internal Task<bool> IsCompleteAsync(Project project, CancellationToken cancellationToken)
+        {
+            // return IsComplete when compilation is not supported. 
+            // regardless whether project support compilation or not, if projectInfo is not complete, we can't gurantee its reference completeness
+            return project.SupportsCompilation
+                ? this.GetCompilationTracker(project.Id).IsCompleteAsync(this, cancellationToken)
+                : project.IsComplete ? SpecializedTasks.True : SpecializedTasks.False;
```

Comment:
Those names work for me.

Important question:  do we need both?

To me, it seems like we could just have one, and it is the transitive one.  i.e. we have something called HasLoadedSucessfully, and it returns true if this projects and all dependents were able to load without issue, false otherwise.

Is there a client that only cares if *this* specific project (and no dependencies) loaded successfully?  If so, which client is that?  I want to know so i can understand why we're need both APIs.